services/device-registry-base/src/main/java/org/eclipse/hono/deviceregistry/service/credentials/AbstractCredentialsService.java

```diff
@@ -165,6 +166,8 @@ protected final CacheDirective getCacheDirective(final String type, final long m
                                             authId,
                                             clientContext,
                                             span);
+                                } else if (credentialsResult.isError()) {
+                                    return Future.succeededFuture(credentialsResult);
```

Comment:
would be good to also have a unit test for this, wouldn't it?

Reply:
I've added a test.